# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../cgit/head.patch
# Copyright (C) 2008 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

diff --git a/ui-tree.c b/ui-tree.c
index 79332fc..b24f693 100644
--- a/ui-tree.c
+++ b/ui-tree.c
@@ -38,6 +38,8 @@ static void print_object(const unsigned char *sha1, char *path)
 	html(" (");
 	cgit_plain_link("plain", NULL, NULL, ctx.qry.head,
 		        curr_rev, path);
+	html(") (");
+	cgit_tree_link("HEAD", NULL, NULL, ctx.qry.head, NULL, path);
 	htmlf(")<br/>blob: %s", sha1_to_hex(sha1));
 
 	html("<table summary='blob content' class='blob'>\n");
